src/contracts/xMETRIC.sol

```diff
@@ -21,13 +21,22 @@ contract xMETRIC is ERC20("xMETRIC", "xMETRIC"), Ownable {
         address to,
         uint256 amount
     ) public override transactor returns (bool) {
-        address spender = _msgSender();
-        _spendAllowance(from, spender, amount);
-        _transfer(from, to, amount);
+        _spendAllowance(from, msg.sender, amount);
```

Comment:
I guess this slightly confuses me now - so to clarify
first we need addTransactor needs to be called before calling this which is essentially whitelisting an address - we check to see if msg.sender is whitelisted first
then we check to see if the from address, owner of current token has enough token which is passed in the amount param in their wallet.
then this contract just mints/creates new xMetric "amount" param and sends it to the "to" param address. 
From there we remove/burn the "amount" param of tokens in the from address?
Basically anyone who is a transactor can call this on behalf of someone else which seems odd
Also is there an easier way to do this?  Maybe not but took a minute for me to grasp

Reply:
My understanding was this:
- We create a monopoly money token (one without an initial or capped supply) that only certain addresses are able to transact (the transactors).
- We add an address as transactor, this address is now allowed to access both transfer and transferFrom functionality.
- As discussed, we replace _transfer with _mint in order to send around arbitrary amounts of tokens.

The use-cases for the function are as followed:
- A transactor (which in my understanding will be accounts controlled/permitted by us) can "transfer" tokens to any user. This directly mints the tokens to the user. We do this because we don't initially distribute supply to ourselves. The receiver of the tokens we could call Alice. 
- Alice can't transfer these tokens as she is not a transactor. However, we want to also be able to test masterchef functionality with these xMetric tokens for instance. So what Alice can do is approve a masterchef contract, where the master contract is added as transactor by the owner. The masterchef contract can then access Alice's funds by using TransferFrom. Here we mimic a transfer by "transfering" the tokens from Alice to the masterchef contract for instance. This is done by decreasing Alice's token balance (by burning) while also decreasing the allowance that Alice has set for the masterchef contract, and increasing the masterchef balance (by minting). This lets us verify the accounting and mimic the accounting.

Additionally, we added burn functionality where any user can burn their own tokens, and the owner can burn anyones tokens.

And it should only be possible for a transactor to spend balance of a user that first approved the transactor to spend set balance, otherwise it will still revert with ← "ERC20: insufficient allowance"

<img width="1000" alt="Screenshot 2022-07-06 at 18 49 21" src="https://user-images.githubusercontent.com/36921585/177602483-c0d886e9-f596-4e9c-93e7-40c8f341d302.png">
.